src/parsers/manifestjson.js

```diff
@@ -285,14 +285,24 @@ export default class ManifestJSONParser extends JSONParser {
     // Not sure about FTP here but CSP spec treats ws/wss as
     // equivalent to http/https.
     const validProtocols = ['ftp:', 'http:', 'https:', 'ws:', 'wss:'];
-    const candidates = ['script-src', 'default-src', 'worker-src'];
+    // The order is important here, 'default-src' needs to be
+    // before 'script-src'
+    const candidates = ['default-src', 'script-src', 'worker-src'];
 
+    let insecureDefaultSrc = false;
+    let secureScriptSrc = false;
     for (let i = 0; i < candidates.length; i++) {
       /* eslint-disable no-continue */
       const candidate = candidates[i];
       if (Object.prototype.hasOwnProperty.call(directives, candidate)) {
         const values = directives[candidate];
 
+        // If the 'default-src' is insecure, check whether the 'script-src'
+        // makes it secure, ie 'script-src: self;'
+        if (insecureDefaultSrc && values.length === 1 && values[0] === '\'self\'') {
+          secureScriptSrc = true;
+        }
```

Comment:
I feel like this is missing a check for `candidate` since this doesn't work in case `worker-src: self;` is being set (for whatever reason)

Reply:
I'd also not use two different variables but rather just use `script-src` to overwrite any insecurity that got set earlier. For example, define a `insecureSrcDirective` which get's set in case we discover an insecure `default-src` and unset in case `script-src` fixes it again.

Reply:
Good catch, thank you!